src/main/java/com/myhexin/cragent/coderabbittest/Counter.java

```diff
@@ -0,0 +1,29 @@
+package com.myhexin.cragent.coderabbittest;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class Counter {
+
+    private AtomicInteger current = new AtomicInteger();
+
+    private final int sampleRate;
+
+    private final int checkValue;
+
+    public Counter(int sampleRate, int checkValue) {
+        this.sampleRate = sampleRate;
+        this.checkValue = checkValue;
+    }
+
+    public boolean checkAndInc() {
+        // 原子获取v
+        int v = current.getAndAccumulate(1, (pre, up) -> {
+            int next = pre + up;
+            if (next >= sampleRate) {
+                return 0;
+            }
+            return next;
+        });
+        return v == checkValue;
+    }
```

Comment:
**Optimization Issue**: The atomic operation inside checkAndInc uses getAndAccumulate in a way that could be simplified to improve readability and possibly performance by directly using incrementAndGet or addAndGet. <br> **Fix**: Simplify the atomic operation to use a more direct method for incrementing and checking the value, which could improve the performance and readability of the code. <br> **Code Suggestion**: 
 ```
 public boolean checkAndInc() {
     int v = current.incrementAndGet();
     if (v >= sampleRate) {
         current.set(0);
         v = 0;
     }
     return v == checkValue;
 }
 ```